### ISSUE TITLE
cannot delete deposit locations

### ISSUE THREAD

**Issue by smutniak** [TexasDigitalLibrary/Vireo#1537]
dragging to 'remove items by...' box brings up intermodal but 'Confirm' button has no effect.  Cancel button works though.

**Comment by kaladay** [comment]
This appears to be the problem and a comment already existing about it:
- https://github.com/TexasDigitalLibrary/Vireo/blob/dff6a55c064205c64574431e70169dc10f0fea01/src/main/java/org/tdl/vireo/controller/DepositLocationController.java#L63

There is also a problem here:
- https://github.com/TexasDigitalLibrary/Vireo/blob/dff6a55c064205c64574431e70169dc10f0fea01/src/main/java/org/tdl/vireo/controller/DepositLocationController.java#L54

**Comment by kaladay** [comment]
I tested this by manually enterring fake data into the DB like this:
```
insert into deposit_location (collection, position, depositor_name, name, password, repository, timeout, username, packager_id)
values ('collection', 1, 'SWORDv1Depositor', 'example', 'a3C/G/UA/iv6xOWhNYs1ug==', 'http://localhost/nowhere', 600, 'user', 1);
```

Delete and update works properly.
If I remove the (optional) `position` column, then the delete fails.
I think the problem may be that the `position` column could be NULL or have bad data.
In this case the solution is to update the database data, replacing the `position` value with a valid number (whole numbers >= 1).

I will be trying to reproduce this with actual data to confirm/deny if this works or not once bad data is removed.

The above comments about "...endpoint is broken..." appears to be a red-herring.

**Comment by kaladay** [comment]
Update from @smutniak:
> It works beautifully for creating a new deposit location but when I drag/drop to delete a deposit_location I get an error:
> ```
> org.postgresql.util.PSQLException: ERROR: update or delete on table "abstract_packager" violates foreign key constraint "fka8yk6a0eicryh80qafkp0shjs" on table "deposit_location"
>   Detail: Key (id)=(1) is still referenced from table "deposit_location".
> ```

Looks like the delete process is already overridden or needs to be overridden to handle deleting the foreign keys first.

**Comment by kaladay** [comment]
There are several problems here.
1. The problem in the above comment, now solved here: #1812
2. The delete fails if the form is not fully filled out (The backend allows for **on behalf of** to be optional but the UI requires it).
3. A constraint violation on delete happens once (1) and (2) above are resolved.

```
org.postgresql.util.PSQLException: ERROR: update or delete on table "abstract_packager" violates foreign key constraint "fka8yk6a0eicryh80qafkp0shjs" on table "deposit_location"
```

A custom delete operation needs to be performed that addresses the foreign key issue.
Why is it trying to delete the abstract_packager?
More than likely there is `CASCADE ON DELETE` of some sort happening.
I will follow up with an investigation and/or PR if I am able to identify the details of problem.
